### PR TITLE
List collections with WiredTiger

### PR DESCRIPTION
**Pull request by cchantep**
Implementation for #284

### CODE REVIEW

File: driver/src/main/scala/api/database.scala [hunk 1]
```diff
@@ -127,27 +127,8 @@ trait DBMetaCommands {
   /** Returns an index manager for this database. */
   def indexesManager(implicit ec: ExecutionContext) = new IndexesManager(self)
 
-  private lazy val collectionNameReader =
```
Comment (cchantep) [inline driver/src/main/scala/api/database.scala]: @sgodbillon Let me know what you think about moving code in this way, to `DefaultDB` (to be able to use `runCommand` with `{ 'listCollections': 1 }`).

File: driver/src/main/scala/api/database.scala [hunk 2]
```diff
@@ -140,13 +144,14 @@ trait DBMetaCommands {
...
   def collectionNames(implicit ec: ExecutionContext): Future[List[String]] = {
-    // TODO: Use command { 'listCollections': 1 }, check with WT
-    collection("system.namespaces").as[BSONCollection]()
+    val wireVer = connection.metadata.map(_.maxWireVersion)
+
+    if (wireVer.exists(_ == MongoWireVersion.V30)) {
+      Command.run(BSONSerializationPack)(self, ListCollectionNames).map(_.names)
+    } else collection("system.namespaces").as[BSONCollection]()
       .find(BSONDocument(
         "name" -> BSONRegex("^[^\\$]+$", "") // strip off any indexes
```
Comment (fehmicansaglam) [inline driver/src/main/scala/api/database.scala]: Are you sure that this regex is correct?

Comment (cchantep) [inline driver/src/main/scala/api/database.scala]: The tests about that on MongoDB < 3 are (still) passing with it. Have you seen any issue about?

Comment (fehmicansaglam) [inline driver/src/main/scala/api/database.scala]: No, but if I am not mistaken that regex means `any name that does not contain $`. I think it should be `any name which does not start with $`.

**Pull request merged**